framework/tst/dslabs/framework/testing/junit/BaseJUnitTest.java

```diff
@@ -343,16 +343,27 @@ protected final boolean goalFound() {
 
     protected final SearchState goalMatchingState() {
         assert !searchResults.goalsSought().isEmpty();
-        assertGoalFound(true);
+        assertGoalFound(null, true);
         return searchResults.goalMatchingState();
     }
 
-    protected final void assertGoalFound() {
-        assert !searchResults.goalsSought().isEmpty();
-        assertGoalFound(false);
+    protected final SearchState findGoalMatchingStateFrom(SearchState start) {
+        return findGoalMatchingStateFrom(start, searchSettings);
+    }
+
+    protected final SearchState findGoalMatchingStateFrom(SearchState start,
+                                                          SearchSettings settings) {
+        bfs(start, settings);
+        assertGoalFound(start, true);
+        return searchResults.goalMatchingState();
+    }
+
+    protected final void assertGoalReachableFrom(SearchState start) {
+        bfs(start);
+        assertGoalFound(start, false);
     }
 
-    private void assertGoalFound(boolean endTestOnFailure) {
+    private void assertGoalFound(SearchState start, boolean endTestOnFailure) {
```

Comment:
I haven't thought carefully about this yet, but as long as sub-classes of `BaseJUnitTest` are calling `bfs` and `dfs` to start searches, shouldn't we _know_ the starting state and search settings used to get there? Maybe `BaseJUnitTest` could just cache those, and you wouldn't need to modify any of the tests.

Reply:
Changed.

Initially I was concerned that it may be confusing to have `initSearchState` and `bfsStartState`, but I'm fine either way.